enhancements/ocp-coreos-layering.md

```diff
@@ -103,23 +109,117 @@ A future enhancement will help automate the above.  For example, we may support
+As we move towards having users manage many clusters (10, 100 or more), it will make sense to support building a node image centrally. This will allow submitting the image to a security scanner or review by a security team before deploying to clusters.
 
-Acme Corp has 300 clusters distributed across their manufacturing centers.  They want to centralize their build system in their main data center, and just distribute those images to single node edge machines.  They provide a `custom-coreos-imagestream` object at installation time, and their node CoreOS image is deployed during the installation of each cluster without a build operation.
+Acme Corp has 300 clusters distributed across their manufacturing centers. They want to centralize their build system in their main data center, and just distribute those images to single node edge machines. They provide a `custom-coreos-imagestream` object at installation time, and their node CoreOS image is deployed during the installation of each cluster without a build operation.
 
 In the future, we may additionally support a signature mechanism (IMA/fs-verity) that allows these images can be signed centrally.  Each node verifies this image rather than generating a custom state.
 
 (Note some unanswered questions below)
 
-### API Extensions
-
-We will continue to support MachineConfig.  The exact details of the mechanics of custom builds are still TBD.
+### External image build workflow
+
+The administrator will see a number of documented example Dockerfiles, like https://github.com/coreos/coreos-layering-examples/
+
+They can copy and extend these and build them using any container build system they like.
+The administrator can also use non-Dockerfile build systems such as Tekton or buildah.
+
+The result will be a container image that can be passed via an API to clusters, in a similar way as way user workloads (pods).
+
+### API Extensions ([MCO-280](https://issues.redhat.com/browse/MCO-280))
+
+Note we will continue to support MachineConfig.  The key goal is to allow the user to override the rhel-coreos base image.
+
+#### Current default proposal: Configure via imagestream
+
+This is currently prototyped in the https://github.com/openshift/machine-config-operator/tree/layering branch.  Here there’s an imagestream per MachineConfigPool, and the administrator can specify an image which overrides the default rhel-coreos base image.
+
+##### User workflow in layering model
+Using layering model, users will be able to build, test and use a custom image based on their requirements.
+The user experience is described in the sub-sections below:
+
+###### Create Customer Base Image
+- First of all the user will create a Customer Base Image. This image can run on compatible OCP clusters such as HyperShift, Heterogeneous cluster, Single Node OpenShift.
+- Create a Dockerfile that uses compatible rhcos image and add the custom content (e.g. rpm, files) on top of it:
```

Comment:
It would help if each step in this workflow was explicit about who the actor is (the user, a software component, etc.) It can be a bit tedious to write that way, but the results are much clearer for a reader to follow.

Reply:
Thanks for the feedback, will make the actor explicit in second pass.